roles/ceph-dashboard/tasks/configure_dashboard.yml

```diff
@@ -211,12 +193,31 @@
       with_items: "{{ groups[iscsi_gw_group_name] }}"
       run_once: true
 
-- name: inject grafana dashboard layouts
-  command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard grafana dashboards update"
-  delegate_to: "{{ groups[mon_group_name][0] }}"
-  run_once: true
-  changed_when: false
-  when: containerized_deployment | bool
+- name: set dashboard endpoints and inject layouts
+  block:
+  - name: set grafana url
+    command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-grafana-api-url {{ dashboard_protocol }}://{{ grafana_server_addr }}:{{ grafana_port }}"
+    delegate_to: "{{ groups[mon_group_name][0] }}"
+    run_once: true
+    changed_when: false
+
+  - name: set alertmanager host
+    command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-alertmanager-api-host {{ dashboard_protocol }}://{{ grafana_server_addr }}:{{ alertmanager_port }}"
+    delegate_to: "{{ groups[mon_group_name][0] }}"
+    run_once: true
+    changed_when: false
+
+  - name: set prometheus host
+    command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard set-prometheus-api-host {{ dashboard_protocol }}://{{ grafana_server_addr }}:{{ prometheus_port }}"
+    delegate_to: "{{ groups[mon_group_name][0] }}"
+    run_once: true
+    changed_when: false
+
+  - name: inject grafana dashboard layouts
+    command: "{{ container_exec_cmd }} ceph --cluster {{ cluster }} dashboard grafana dashboards update"
+    delegate_to: "{{ groups[mon_group_name][0] }}"
+    run_once: true
```

Comment:
you can probably set run_once for the entire block

Reply:
++